tests/Aspire.Hosting.Tests/Dashboard/DashboardServiceTests.cs

```diff
@@ -130,6 +124,46 @@ await resourceNotificationService.WaitForResourceAsync(testResource.Name, r =>
         await CancelTokenAndAwaitTask(cts, task);
     }
 
+    [Fact]
+    public async Task CreateResource_NoChild_WithoutHealthChecks_WithInitialSnapshot_HealthStatusReportedCorrectly()
+    {
+         // Arrange
+        var resourceLoggerService = new ResourceLoggerService();
+        var resourceNotificationService = new ResourceNotificationService(NullLogger<ResourceNotificationService>.Instance, new TestHostApplicationLifetime(), new ServiceCollection().BuildServiceProvider(), resourceLoggerService);
+        await using var dashboardServiceData = new DashboardServiceData(resourceNotificationService, resourceLoggerService, NullLogger<DashboardServiceData>.Instance, new DashboardCommandExecutor(new ServiceCollection().BuildServiceProvider()));
+        var dashboardService = new DashboardService(dashboardServiceData, new TestHostEnvironment(), new TestHostApplicationLifetime(), NullLogger<DashboardService>.Instance);
+
+        var testResource = new TestResource("test-resource");
+        using var builder = TestDistributedApplicationBuilder.Create();
+        builder.AddResource(testResource)
+            .WithInitialState(new CustomResourceSnapshot
+            {
+                ResourceType = string.Empty,
+                Properties = [],
+                HealthStatus = HealthStatus.Degraded,
+            });
+
+        await resourceNotificationService.PublishUpdateAsync(testResource, s => { return s with { State = new ResourceStateSnapshot("Running", null) }; });
+
+        var cts = new CancellationTokenSource();
+        var context = TestServerCallContext.Create(cancellationToken: cts.Token);
+        var writer = new TestServerStreamWriter<WatchResourcesUpdate>(context);
+
+        // Act
+        var task = dashboardService.WatchResources(
+            new WatchResourcesRequest(),
+            writer,
+            context);
+
+        // Assert
+        var initialRead = await writer.ReadNextAsync();
+        var initialSnapshot = Assert.Single(initialRead.InitialData.Resources);
```

Comment:
This will be flaky. There is no guarentee that published resource will be in the initial data returned by the service.

Waiting for the resource mostly fixes it (see above test). However I believe you need to wait on another type to properly fix it. See https://github.com/dotnet/aspire/pull/6366 and the changes I made on the `DashboardDataService`.

Reply:
so is this dependent on #6366?

Reply:
You can do what the test above is currently doing. It's 99.9% stable. Create an issue to update it once 6366 is merged.